lib/sdotypes.c

```diff
@@ -2930,58 +2951,114 @@ sdo_rendezvous_list_t *sdo_rendezvous_list_alloc(void)
 void sdo_rendezvous_list_free(sdo_rendezvous_list_t *list)
 {
 	sdo_rendezvous_t *entry, *next;
+	sdo_rendezvous_directive_t *directive_entry, *directive_next;
 
 	if (list == NULL) {
 		return;
 	}
 
 	/* Delete all entries. */
-	next = entry = list->rv_entries;
-	while (entry) {
-		next = entry->next;
-		sdo_rendezvous_free(entry);
-		entry = next;
-	};
-
-	list->num_entries = 0;
+	directive_entry = directive_next = list->rv_directives;
+	while (directive_entry) {
+		next = entry = directive_entry->rv_entries;
+		while (entry) {
+			next = entry->next;
+			sdo_rendezvous_free(entry);
+			entry = next;
+		};
+		directive_next = directive_entry->next;
+		directive_entry = directive_next;
+	}
+	list->num_rv_directives = 0;
 	sdo_free(list);
 }
 
 /**
- * Add the rendzvous to the rendzvous list
- * @param list - pointer to the rendzvous list
- * @param rv - pointer to the rendezvous to be added to the list
+ * Add the RendezvousDirective to the RendezvousInfo list
+ * @param list - pointer to the RendezvousInfo list
+ * @param rv - pointer to the RendezvousDirective to be added to the list
  * @return number of entries added if success else error code
  */
-int sdo_rendezvous_list_add(sdo_rendezvous_list_t *list, sdo_rendezvous_t *rv)
+bool sdo_rendezvous_directive_add(sdo_rendezvous_list_t *list,
+	sdo_rendezvous_directive_t *directive) {
+	if (list == NULL || directive == NULL)
+		return 0;
```

Comment:
better to define constants for true and false and use those instead of 0 directly

Reply:
This was supposed to return 'int' representing number of entries. Updated the method return type from 'bool' to 'int'.